In [2]:
import pandas as pd

# 1. Cargar los datos (ajusta el nombre del archivo si es distinto)
df = pd.read_csv('dataset/churn.csv')

In [3]:
# 2. Un vistazo rápido a las primeras 5 filas para entender la estructura
print("--- PRIMERAS 5 FILAS ---")
display(df.head()) 

--- PRIMERAS 5 FILAS ---


,RowNumber,CustomerId,Surname,CreditScore,Geography,Gender,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
0,1,15634602,Hargrave,619,France,Female,42,2,0.00,1,1,1,101348.88,1
1,2,15647311,Hill,608,Spain,Female,41,1,83807.86,1,0,1,112542.58,0
2,3,15619304,Onio,502,France,Female,42,8,159660.80,3,1,0,113931.57,1
3,4,15701354,Boni,699,France,Female,39,1,0.00,2,0,0,93826.63,0
4,5,15737888,Mitchell,850,Spain,Female,43,2,125510.82,1,1,1,79084.10,0


## Detalles de campos

- RowNumber: Numero de filas (no aporta nada)
- CustomerId: Identificador de cliente (parece tener 8 digitos)
- Surname: Apellido del cliente
- CreditScore: Puntaje de credito
- Geography: La ubicacion del cliente
- Gender: El genero del cliente
- Age: La edad del cliente
- Tenure: Numero de años que ha sido cliente del banco
- Balance: Saldo del cliente
- NumOfProducts: Numero de productos que tiene el cliente con el banco
- HasCrCard: Indica si el cliente tiene tarjeta de credito
- isActiveMember: Indica si el cliente es miembro activo del banco
- EstimatedSalary: Sueldo estimado del cliente
- Exited: Indica si el cliente se fue del banco

In [4]:
# 3. Información general (Vital para ver si hay nulos y qué tipo de datos tenemos)
print("\n--- INFORMACIÓN DEL DATASET ---")
df.info()


--- INFORMACIÓN DEL DATASET ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   RowNumber        10000 non-null  int64  
 1   CustomerId       10000 non-null  int64  
 2   Surname          10000 non-null  object 
 3   CreditScore      10000 non-null  int64  
 4   Geography        10000 non-null  object 
 5   Gender           10000 non-null  object 
 6   Age              10000 non-null  int64  
 7   Tenure           10000 non-null  int64  
 8   Balance          10000 non-null  float64
 9   NumOfProducts    10000 non-null  int64  
 10  HasCrCard        10000 non-null  int64  
 11  IsActiveMember   10000 non-null  int64  
 12  EstimatedSalary  10000 non-null  float64
 13  Exited           10000 non-null  int64  
dtypes: float64(2), int64(9), object(3)
memory usage: 1.1+ MB


Ningun campo tiene valores nulos.
Pero se observa que los tipos de datos no son adecuados para algunos campos (usa mucha memoria)

In [5]:
# 4. Estadísticas descriptivas (Para ver promedios de edad, sueldos, saldos)
print("\n--- ESTADÍSTICAS DESCRIPTIVAS ---")
display(df.describe())


--- ESTADÍSTICAS DESCRIPTIVAS ---


,RowNumber,CustomerId,CreditScore,Age,Tenure,Balance,NumOfProducts,HasCrCard,IsActiveMember,EstimatedSalary,Exited
count,10000.00000,1.000000e+04,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.00000,10000.000000,10000.000000,10000.000000
mean,5000.50000,1.569094e+07,650.528800,38.921800,5.012800,76485.889288,1.530200,0.70550,0.515100,100090.239881,0.203700
std,2886.89568,7.193619e+04,96.653299,10.487806,2.892174,62397.405202,0.581654,0.45584,0.499797,57510.492818,0.402769
min,1.00000,1.556570e+07,350.000000,18.000000,0.000000,0.000000,1.000000,0.00000,0.000000,11.580000,0.000000
25%,2500.75000,1.562853e+07,584.000000,32.000000,3.000000,0.000000,1.000000,0.00000,0.000000,51002.110000,0.000000
50%,5000.50000,1.569074e+07,652.000000,37.000000,5.000000,97198.540000,1.000000,1.00000,1.000000,100193.915000,0.000000
75%,7500.25000,1.575323e+07,718.000000,44.000000,7.000000,127644.240000,2.000000,1.00000,1.000000,149388.247500,0.000000
max,10000.00000,1.581569e+07,850.000000,92.000000,10.000000,250898.090000,4.000000,1.00000,1.000000,199992.480000,1.000000


En esta informacion no se debe considerar los campos:
- RowNumber
- CustomerId
- HasCrCard
- IsActiveMember
- Exited

In [6]:
# 5. La métrica de negocio: ¿Cuántos se fueron (1) vs cuántos se quedaron (0)?
print("\n--- TASA DE FUGA ACTUAL (CHURN RATE) ---")
# normalize=True nos da el porcentaje en lugar del conteo
print(df['Exited'].value_counts(normalize=True) * 100)


--- TASA DE FUGA ACTUAL (CHURN RATE) ---
0    79.63
1    20.37
Name: Exited, dtype: float64


De los 10000 clientes el 79.63% mantiene relacion con el banco y el 20.37% se van del banco.

In [7]:
# 1. Eliminar columnas que no aportan valor analítico (ruido)
# Quitamos Surname por privacidad de datos (buena práctica) y los IDs que no sirven para el modelo
df_clean = df.drop(columns=['RowNumber', 'CustomerId', 'Surname'])

In [8]:
# 2. Optimizar memoria y corregir tipos de datos
# Convertir texto a 'category' reduce drásticamente el peso en RAM
df_clean['Geography'] = df_clean['Geography'].astype('category')
df_clean['Gender'] = df_clean['Gender'].astype('category')

In [9]:
# Convertir variables binarias para que no se mezclen en la estadística continua
df_clean['HasCrCard'] = df_clean['HasCrCard'].astype('category')
df_clean['IsActiveMember'] = df_clean['IsActiveMember'].astype('category')
# Dejaremos 'Exited' como entero (0 y 1) porque nos servirá para sumar rápidamente la fuga en Power BI y SQL

In [10]:
# 3. Comprobar la optimización de memoria
print("--- NUEVA INFORMACIÓN (MEMORIA OPTIMIZADA) ---")
df_clean.info()

--- NUEVA INFORMACIÓN (MEMORIA OPTIMIZADA) ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype   
---  ------           --------------  -----   
 0   CreditScore      10000 non-null  int64   
 1   Geography        10000 non-null  category
 2   Gender           10000 non-null  category
 3   Age              10000 non-null  int64   
 4   Tenure           10000 non-null  int64   
 5   Balance          10000 non-null  float64 
 6   NumOfProducts    10000 non-null  int64   
 7   HasCrCard        10000 non-null  category
 8   IsActiveMember   10000 non-null  category
 9   EstimatedSalary  10000 non-null  float64 
 10  Exited           10000 non-null  int64   
dtypes: category(4), float64(2), int64(5)
memory usage: 586.6 KB


In [11]:
# 4. Ver estadísticas descriptivas (¡Ahora sí, solo de datos reales!)
print("\n--- ESTADÍSTICAS NUMÉRICAS LIMPIAS ---")
display(df_clean.describe())


--- ESTADÍSTICAS NUMÉRICAS LIMPIAS ---


,CreditScore,Age,Tenure,Balance,NumOfProducts,EstimatedSalary,Exited
count,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000,10000.000000
mean,650.528800,38.921800,5.012800,76485.889288,1.530200,100090.239881,0.203700
std,96.653299,10.487806,2.892174,62397.405202,0.581654,57510.492818,0.402769
min,350.000000,18.000000,0.000000,0.000000,1.000000,11.580000,0.000000
25%,584.000000,32.000000,3.000000,0.000000,1.000000,51002.110000,0.000000
50%,652.000000,37.000000,5.000000,97198.540000,1.000000,100193.915000,0.000000
75%,718.000000,44.000000,7.000000,127644.240000,2.000000,149388.247500,0.000000
max,850.000000,92.000000,10.000000,250898.090000,4.000000,199992.480000,1.000000


In [12]:
# 5. Guardar el archivo limpio para la Fase SQL
df_clean.to_csv('Bank_Churn_Cleaned.csv', index=False)
print("\n✅ Archivo 'Bank_Churn_Cleaned.csv' exportado con éxito y listo para la base de datos.")


✅ Archivo 'Bank_Churn_Cleaned.csv' exportado con éxito y listo para la base de datos.
